pkgs/stdenv/generic/get-env.sh.nix

```diff
@@ -0,0 +1,136 @@
+builtins.toFile "get-env.sh" ''
+
+  set -e
+  if [ -e .attrs.sh ]; then source .attrs.sh; fi
+
+  export IN_NIX_SHELL=impure
+  export dontAddDisableDepTrack=1
+
+  if [[ -n $stdenv ]]; then
+      source $stdenv/setup
+  fi
+
+  # Better to use compgen, but stdenv bash doesn't have it.
+  __vars="$(declare -p)"
+  __functions="$(declare -F)"
```

Comment:
But also, I have barely any clue what this shell script even does. Do we really need everything it does or is this complexity justified? It should have a lot more comments.

Reply:
Well actually it looks like we can pull it from nix master:
https://github.com/NixOS/nix/blob/master/src/nix/get-env.sh

The version in my local tree still has this in a header, but this copy wouldn't require any IFD.